src/patch/peripheral.rs

```diff
@@ -749,6 +905,27 @@ impl PeripheralExt for Peripheral {
             }
         }
 
+        if let Some(expand_cluster) = pmod.get_yaml("_expand_cluster") {
+            match expand_cluster {
+                Yaml::String(cspec) => {
+                    self.expand_cluster(cspec, &ppath)
+                        .with_context(|| format!("During expand of `{cspec}` cluster"))?;
+                }
+                Yaml::Array(clusters) => {
+                    for cspec in clusters {
+                        let cspec = cspec.str()?;
+                        self.expand_cluster(cspec, &ppath)
+                            .with_context(|| format!("During expand of `{cspec}` cluster"))?;
+                    }
+                }
+                _ => {
+                    return Err(anyhow!(
+                        "`_expand_cluster` requires string value or array of strings"
+                    ))
+                }
+            }
+        }
+
```

Comment:
This can be shortened to:
```suggestion
        for cspec in pmod.str_vec_iter("_expand_cluster").map_err(|_| anyhow!("`_expand_cluster` requires string value or array of strings"))? {
            let cspec = cspec.str()?;
            self.expand_cluster(cspec, &ppath)
                .with_context(|| format!("During expand of `{cspec}` cluster"))?;
        }
```

Reply:
Although I'm not sure. Should there be a list of strings or a hashmap.
Do we plan to pass additional any information in future?

Reply:
I did this the in the longer form originally because I thought that one might want to pass an array of suffixes, to replace automatic _1, _2 scheme, but I think that it would make more sense for the user to modify dim_index before expanding cluster, since that is now implemented. 
I also was thinking that one might want to pass an argument to expand the cluster into a register array, but I think that might be better served by a separate _cluster_to_array function.

Reply:
I actually think that adding the option to prefix all expanded cluster objects here might be very useful. I just ran into a situation where a cluster member has the same name as another register in the peripheral